### PR TITLE
Add HIP symbol redefinitions

### LINKED ISSUE #5348: Need housekeeping macros for HIP
According to an earlier discussion (https://github.com/cupy/cupy/pull/4367#discussion_r542051342), it's better to collect the macro redefinitions for functions that are not supported by HIP (ex: `__shfl*sync()`) in one place so that they are always correctly included when compiling on HIP. This would benefit CuPy JIT too (ex: #5335). I could try to work on this later next week...

### PR DESCRIPTION
**Pull request by leofang**
Close #5348.

### CODE REVIEW

File: cupyx/jit/_builtin_funcs.py [hunk 1]
```diff
@@ -126,8 +126,9 @@ def __call__(self, *, mask=0xffffffff):
 
     def call(self, env, *, mask=None):
         if runtime.is_hip:
-            warnings.warn(f'mask {mask} is ignored on HIP', RuntimeWarning)
-            mask = None
+            if mask is not None:
+                warnings.warn(f'mask {mask} is ignored on HIP', RuntimeWarning)
+                mask = None
```
Comment (leofang) [inline cupyx/jit/_builtin_funcs.py:131]: This fixes my overlook in #5335 😅

**Comment by leofang** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
The Windows failure seems to be related to #5364.

**Comment by leofang** [comment]
@emcastillo It passed on all CIs!

**Comment by leofang** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]

**Comment by leofang** [comment]
Hmm this is already backported.

**Comment by kmaehashi** [comment]
Added `backport` label to #5475 so that bot will no longer warn 😄

**Comment by leofang** [comment]
Thanks!

**Pull request merged**